File: services_go/utils/apiResponse.go
package utils

import (
	"github.com/gin-gonic/gin"
)

func MakeResponse(message string, data interface{}, detail string) gin.H {
	// Chuyển đổi data thành một bản đồ
	dataMap, ok := data.(map[string]interface{})

	var response gin.H
	if !ok {
		response = gin.H{"message": message, "detail": detail, "data": data}
	} else {
		response = gin.H{"message": message, "detail": detail}
		for key, value := range dataMap {
			response[key] = value
		}
	}

	return response
}

File: services_python/utils/api_response.py
def make_response(
    message: str = None,
    detail: str = None,
    data=None,
    total: int = None,
    skip: int = None,
    limit: int = None,
):
    response = {
        "message": message,
        "detail": detail,
        "data": data,
        "total": total,
        "skip": skip,
        "limit": limit,
    }

    # Filter out key-value pairs where the value is None
    response = {k: v for k, v in response.items() if v is not None}

    return response
